Firestore/Source/API/FIRFirestore.mm

```diff
@@ -236,6 +236,35 @@ - (void)ensureClientConfigured {
       FSTAssert(_settings.host, @"FirestoreSettings.host cannot be nil.");
       FSTAssert(_settings.dispatchQueue, @"FirestoreSettings.dispatchQueue cannot be nil.");
 
+      if (!_settings.enableTimestampsInSnapshots) {
+        // TODO(varconst): DO NOT SUBMIT: doublecheck these code snippets before submitting.
+        FSTWarn(
+            @"The default behavior for NSDate objects stored in Firestore is going to change "
```

Comment:
FYI- `NSDate` is Obj-C specific while the rest of this warning is Swift-oriented. I'd be inclined to say "timestamps" (which can be read generically or as the actual [Firestore term](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp)) or "system Date objects" (which is my best attempt to reference [NSDate using Swift terminology](https://developer.apple.com/documentation/foundation/date), but feel free to come up with something better).

Feel free to leave the PR as-is for now, but consider incorporating this along with final API review feedback, etc.